### ISSUE TITLE
Unable to connect to insecure repository

### ISSUE THREAD

**Issue by webmutation** [docker/toolbox#446]
I recently upgraded my docker toolbox (1.7.0 > 1.10.2) and jfrog artifactory (3.7.0 > 4.5.2) docker repository and i am now unable to connect in a --insecure-registry manner to push and pull from the repository

[Enviornment]
docker toolbox (1.10.2)
jfrog artifactory (4.5.2)
server is behind corporate proxy not accessible to public 
lower ports are blocked must use higher 8081 and 4443 TLS/SSL

I have a self-signed insecure docker repository on a private network. docker toolbox shell and boot2docker vm (docker-machine ssh default) can talk to the server that has the machine

I have added the appropriate HTTP_PROXY and HTTPS_PROXY settings and NO_PROXY setting to the creation of my docker machine

`"${DOCKER_MACHINE}" create -d virtualbox --virtualbox-memory 2048 --engine-insecure-registry internal-docker.int:8081 --engine-insecure-registry internal-docker.int:4443 --engine-env HTTP_PROXY=http://user:password@NNN.NNN.NNN.NNN:8012 --engine-env HTTPS_PROXY=https://user:password@NNN.NNN.NNN.NNN:8012 --engine-env NO_PROXY=internal-docker.int "${VM}"`

The config file has the appropriate InsecureRegistry 

> "HostOptions": {
>         "Driver": "",
>         "Memory": 0,
>         "Disk": 0,
>         "EngineOptions": {
>             "ArbitraryFlags": [],
>             "Dns": null,
>             "GraphDir": "",
>             "Env": [
>                 "HTTP_PROXY=http://user:password@NNN.NNN.NNN.NNN:8012",
>                 "HTTPS_PROXY=https://user:password@NNN.NNN.NNN.NNN:8012",
>                 "NO_PROXY=internal-docker.int:4443"
>             ],
>             "Ipv6": false,
>             "InsecureRegistry": [
>                 "internal-docker.int:8081",
>                 "internal-docker.int:4443"
>             ],

However I am currently unable to login to my repository in a insecure manner. Whenever i try to do `Docker Login` i get the following returned 

> $ docker login -e "danielmd3000@gmail.com" -u "user" -p "password" https://internalserver.int:4443
> Error response from daemon: invalid registry endpoint https://internalserver.int:4443/v0/: unable to ping registry endpoint https://internalserver.int:4443/v0/
> v2 ping attempt failed with error: Get https://internalserver.int:4443/v2/: x509: certificate signed by unknown authority
>  v1 ping attempt failed with error: Get https://internalserver.int:4443/v1/_ping: x509: certificate signed by unknown authority. If this private registry supports only HTTP or HTTPS with an unknown CA certificate, please add `--insecure-registry internalserver.int:4443` to the daemon's arguments. In the case of HTTPS, if you have access to the registry's CA certificate, no need for the flag; simply place the CA certificate at /etc/docker/certs.d/internalserver.int:4443/ca.crt

However doing curl -k -uUSER:PASSWORD "https://internal-docker.int:4443/v2/auth" returns an authorization token.

> "https://internal-docker.int:4443" : {
>           "auth" : "TRasdwerdsTpBUEJSaWUxWm16OFhNQmlk",
>           "email" : "user@mail.com"
>       }

I add this token to my .docker/config.json that already contains the key for "https://index.docker.io/v1/" that was created by running Docker Login

However when I try to **docker push internalserver.int:4443** to the repo I am unable to do so 

> unknown: Not Found

I have created a similar environment on a cloud instance and I am faced with the same issue, i cannot push or login in a insecure manner. 
1. Does Docker Login try to use http or ignore the certificates if its an insecure registry? (Because it always tries to use  https looking at the error messages.
2. Is it possible to use an InsecureRegistry outside of localhost? Is this a supported scenario?

I have setup other private repositories with success and can push and pull to those.

**Comment by vberruchon** [comment]
Perhaps not the only problem but I can see in your config.json : 
"NO_PROXY=internal-docker.int**:4443**"
I don't think you should add the port in NO_PROXY as it lists the domains you don't want to pass by the proxy.

**Comment by webmutation** [comment]
That can't go thru the proxy, it's internal. Can't be accessed via proxy. It has to be whitelisted with NO_PROXY otherwise  you get error message

**Comment by vberruchon** [comment]
Yes but again you shoudn't put the port in the NO_PROXY, only DNS name or IP.
